src/webview/components/memory-table.tsx

```diff
@@ -133,9 +155,29 @@ namespace MemorySizeOptions {
 export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTableState> {
 
     protected datatableRef = React.createRef<DataTable<MemoryRowData[]>>();
+    protected resizeObserver?: ResizeObserver;
 
-    protected get isShowMoreEnabled(): boolean {
-        return !!this.props.memory?.bytes.length;
+    protected get datatableWrapper(): HTMLElement | undefined {
+        return this.datatableRef.current?.getElement().querySelector<HTMLElement>('[data-pc-section="wrapper"]') ?? undefined;
+    }
+
+    protected get isLoading(): boolean {
+        return this.props.isMemoryFetching;
+    }
+
+    protected get visibleRowsInDatableCount(): number {
+        const wrapper = this.datatableWrapper;
+
+        if (wrapper) {
+            return MemoryTable.approximateNumberOfFittingRows(wrapper);
+        }
+
+        return 1;
+    }
+
+    protected get appendMoreMemoryRowsCount(): number {
+        const buffer = 8;
+        return this.visibleRowsInDatableCount + buffer;
```

Comment:
These mostly appear to be basically single-use getters in one chain of calculation. More transparent and isomorphic to write a function for this.

Reply:
Just one question, because I'm curious: In which cases would you use then getters?